File: Backend/src/main/java/com/espumablanca/backend/application/usecase/PermissionService.java
package com.espumablanca.backend.application.usecase;

import com.espumablanca.backend.domain.model.dto.request.permission.PermissionRequest;
import com.espumablanca.backend.domain.model.dto.response.permission.PermissionDto;
import com.espumablanca.backend.domain.model.dto.response.permission.PermissionInRoleDto;

import java.util.List;

public interface PermissionService {

    PermissionDto create(PermissionRequest request);

    List<PermissionDto> getAll();

    List<PermissionInRoleDto> getAllByRoleId(Long roleId);

}

File: EspublaBlancaFrontend/src/app/core/services/api/permission.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { environment } from '../../../../environments/environment.development';
import { PermissionInRoleDto } from '../../models/permission/response/permission-in-role-dto.model';

@Injectable({
  providedIn: 'root',
})
export class PermissionService {
  private readonly endpoint = environment.apiUrl + '/protected/permissions';
  private readonly http = inject(HttpClient);

  getAllByRoleId(roleId: number) {
    return this.http.get<PermissionInRoleDto[]>(this.endpoint + '/' + roleId);
  }
}
